pages/api/create_class_teacher.js

```diff
@@ -1,6 +1,33 @@
 import prisma from '../../prisma/prisma';
+import { unstable_getServerSession } from 'next-auth';
+import { authOptions } from './auth/[...nextauth]';
 
 export default async function handle(req, res) {
+  //unstable_getServerSession is recommended here: https://next-auth.js.org/configuration/nextjs
+  const session = await unstable_getServerSession(req, res, authOptions);
+
+  if (!req.method == 'POST') {
+    res.status(405).end();
+  }
+
+  if (!session) {
+    res.status(403).end();
+  }
+
+  let users = await prisma.user.findMany({
```

Comment:
See if possible to findunique based on user email, and then check the role of that unique instead.

This code gets a list of ALL teachers, then sees if user is in that list.

Instead, we should get user's data based on findunique(user email), then check if the data we get back has the right role.

This needs to be changed for all endpoints. My bad.